refactor: System endpoint refactoring

amarnath-ac opened the pull request:
- Refactored system endpoint code and added new system handler.
- Add comprehensive unit tests for SystemsHandler with full coverage.

amarnath-ac committed:
refactor: System endpoint refactoring

- Refactored system endpoint code and added new system handler.
- Add comprehensive unit tests for SystemsHandler with 100% coverage.

Signed-off-by: C, Amarnath <amarnath.c@intel.com>

sudhir-intc commented on redfish/internal/controller/http/v1/handler/routes.go:
```diff
@@ -21,29 +21,18 @@ const (
-// Add dependencies here if needed (e.g., usecase, presenter, etc.)
+// RedfishServer implements the Redfish API handlers and delegates operations to specialized handlers
 type RedfishServer struct {
+	SystemsHandler   *SystemsHandler
```
What is the advantage you see in having a new SystemHandler? From the implementation, I see that it introduces an additional layer of abstraction.  Currently this only has the ComputerSystemUC and logger which is already present in the Redfish Server. 
Instead we could move the implementation of all Systems related paths to systems.go
For ex:  func (s *RedfishServer) GetRedfishV1Systems(c *gin.Context) to the systems.go can be declared and implemented in systems.go

amarnath-ac replied:
address comment

amarnath-ac committed:
refactor: fix review comments

- Remove SystemsHandler interface and move GetRedfishV1Systems and
GetRedfishV1SystemsComputerSystemId from routes.go to systems.go
- Refactor test suite to use concrete types instead of mock.

Signed-off-by: C, Amarnath <amarnath.c@intel.com>

sudhir-intc committed:
Merge branch 'main' into rf_system_refactor
